.github/workflows/main_workflow.yml

```diff
@@ -1,19 +1,40 @@
-name: CI
+name: Java CI
 
-on: [pull_request]
+on: [push, pull_request]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
 
     steps:
-    - uses: actions/checkout@v2
+      - uses: actions/checkout@v2
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
```

Comment:
Why don't you just commit gradlew with executable permission instead of doing it each CI run?